src/Sylius/Component/Inventory/Factory/InventoryUnitFactory.php

```diff
@@ -52,7 +35,7 @@ public function create(StockableInterface $stockable, $quantity, $state = Invent
         $units = new ArrayCollection();
 
         for ($i = 0; $i < $quantity; $i++) {
-            $inventoryUnit = $this->repository->createNew();
+            $inventoryUnit = parent::createNew();
```

Comment:
The method is inherited, so we should use `$inventoryUnit = $this->createNew();`.